Changelog v0.4.2 (#461)
hashicorp/consul-terraform-sync: CHANGELOG.md

## 0.4.2 (November 22, 2021)
FEATURES:
* Support TLS and mutual TLS for the CTS API and CLI. [[GH-466](https://github.com/hashicorp/consul-terraform-sync/issues/466)]
* **(Enterprise Only)** Add Terraform Cloud workspace tagging support to add, require, and restrict tags with new `driver.terraform-cloud.workspaces` options.

IMPROVEMENTS:
* **(Enterprise Only)** Add default address for the Terraform Cloud driver to https://app.terraform.io.

BUG FIXES:

> * Fix Services Regex, Catalog Services, and Consul KV conditions to use Consul API blocking queries for more efficient monitoring of changes on Consul. [[GH-460](https://github.com/hashicorp/consul-terraform-sync/pull/460), [GH-467](https://github.com/hashicorp/consul-terraform-sync/pull/467)]

Reviewer: Maybe this could be `Services` rather than `Services Regex` since that's how the condition is documented elsewhere?

```suggestion
* Fix Services, Catalog Services, and Consul KV conditions to use Consul API blocking queries for more efficient monitoring of changes on Consul. [[GH-460](https://github.com/hashicorp/consul-terraform-sync/pull/460), [GH-467](https://github.com/hashicorp/consul-terraform-sync/pull/467)]
```

Reviewer 2: Yea, I get what you're saying - user-facing wise we don't use this language. I think the extra distinction here is that "Services condition" could refer to services condition configured via `task.services` and `condition "services"` and this fix I _believe_ is just for `condition "services"`.

What about:

> Fix conditions that are configured through Services, Catalog Services, and Consul KV condition blocks to use Consul API blocking queries for more efficient monitoring of changes on Consul. [[GH-460](https://github.com/hashicorp/consul-terraform-sync/pull/460), [GH-467](https://github.com/hashicorp/consul-terraform-sync/pull/467)]

Reviewer: sounds good to me!

Author: The default Services condition was working as expected though. The bug and fix only affects the services condition when the regular expression is configured

Reviewer 2: Gotcha, thanks for clarifying Kim. It sounds like having the "Regex" adds necessary details on the Services condition vs. naming a feature "Services Regex condition"

Reviewer: Can we mention specifically that regex needs to be configured then to reproduce the issue?

How about: 

Fix for more efficient monitoring of changes on Consul when specifying conditions using Consul KV and Catalog Services condition blocks and when configuring regex using the Services condition block  [GH-460, GH-467]

Reviewer 2: @wilkermichael and I just chatted about this and are now suggesting:

> Fix Services condition when configured with regex, Consul KV condition, and Catalog Services condition to use Consul API blocking queries for more efficient monitoring of changes on Consul. [GH-460, GH-467]

@findkim does this capture things ok to you?

Author: I'm liking these iterations! Yeah let's go with the latest one

Author: ```suggestion
* Fix Services condition when configured with regex, Consul KV condition, and Catalog Services condition to use Consul API blocking queries for more efficient monitoring of changes on Consul. [[GH-460](https://github.com/hashicorp/consul-terraform-sync/pull/460), [GH-467](https://github.com/hashicorp/consul-terraform-sync/pull/467)]
```

* Fix issue where choosing to cancel when using the Enable CLI still enabled the task. [[GH-451](https://github.com/hashicorp/consul-terraform-sync/issues/451)]
* Fix issue where Update Task API unexpectedly updated the task when running with inspect mode. [[GH-465](https://github.com/hashicorp/consul-terraform-sync/issues/465)]
* Fix panic when there is a Terraform validation warning related to provider blocks when CTS runs Terraform CLI v0.15+. [[GH-473](https://github.com/hashicorp/consul-terraform-sync/issues/473)]

> * **(Enterprise Only)** Fix issue where configured service block filter was not being used to filter monitored service instances when using the Terraform Cloud driver. [[GH-454](https://github.com/hashicorp/consul-terraform-sync/issues/454)]

Reviewer: I forget where we landed on this, but should we having Enterprise github issues created in the OSS repo?

I'm referring to the github issue itself, not to documenting it as part of this changelog.

Reviewer 2: From what I recall, I believe we agreed to create enterprise issues (tagged with the enterprise label) in the oss repo for transparency. We don't want to provide a link that no community member can access in the changelog. Do you have concerns or just double-checking?

Reviewer: Just double checking, all good

> * **(Enterprise Only)** Fix issue when using Terraform Cloud driver with a scheduled task configured with a consul-kv source input did not run at scheduled time when only the monitored services had changed. [[GH-502](https://github.com/hashicorp/consul-terraform-sync/issues/502)]

Reviewer: Minor rephrasing suggestion:

```suggestion
* **(Enterprise Only)** Fix issue where using the Terraform Cloud driver with a scheduled task configured with a consul-kv source input did not run at the scheduled time when only the monitored services had changed. [[GH-502](https://github.com/hashicorp/consul-terraform-sync/issues/502)]
```
